web/packages/teleport/src/Roles/RoleEditor/StandardEditor/Resources.tsx

```diff
@@ -379,6 +382,60 @@ export function KubernetesAccessSection({
   );
 }
 
+function KubernetesResourceKindView({
+  value,
+  validation,
+  isProcessing,
+  onChange,
+  roleVersion,
+}: {
+  value: KubernetesResourceModel;
+  validation: KubernetesResourceValidationResult['kind'];
+  isProcessing: boolean;
+  onChange?(m: KubernetesResourceModel): void;
+  roleVersion: RoleVersion;
+}) {
+  const supportsCRDs = supportsKubernetesCustomResources(roleVersion);
+  const v7selectField = (
+    <FieldSelect
+      label="Kind"
+      isDisabled={isProcessing}
+      options={kubernetesResourceKindOptionsV7.filter(
+        elem => roleVersion == 'v7' || elem.value == 'pod' // In v7, we have the fill list, in v6 and earlier, only pod.
+      )}
+      value={value.kind}
+      rule={precomputed(validation)}
+      onChange={k => onChange?.({ ...value, kind: k })}
+    />
```

Comment:
This looks declarative, but it this is runtime code. Instead of running it all the time and only using it when we don't support CRDs, can we move it into the if statement?

Reply:
Done: https://github.com/gravitational/teleport/pull/54319/commits/51db28d3eaf9b06c9465e90839565fb57e336cd9